cli/commands/benchmark-web-vitals.mjs

```diff
@@ -242,117 +348,170 @@ async function benchmarkURL( browser, params ) {
 		}
 
 		// Set viewport similar to @wordpress/e2e-test-utils 'large' configuration.
-		await page.setViewport( { width: 960, height: 700 } );
+		await page.setViewport( { width: 960, height: 700 } ); // @todo This should be configurable via command options so that mobile viewport can be loaded.
 		await page
 			.mainFrame()
 			.waitForFunction(
 				'window.innerWidth === 960 && window.innerHeight === 700'
 			);
 
 		// Load the page.
-		const url = new URL( params.url );
-		url.searchParams.append( 'rnd', String( Math.random() ) );
+		const urlObj = new URL( url );
+		urlObj.searchParams.append( 'rnd', String( Math.random() ) );
 
 		// Make sure any username and password in the URL is passed along for authentication.
-		if ( url.username && url.password ) {
+		if ( urlObj.username && urlObj.password ) {
 			await page.authenticate( {
-				username: url.username,
-				password: url.password,
+				username: urlObj.username,
+				password: urlObj.password,
 			} );
 		}
 
-		const response = await page.goto( url.toString(), {
+		const response = await page.goto( urlObj.toString(), {
 			waitUntil: 'networkidle0',
 		} );
-		await page.addScriptTag( { content: scriptTag, type: 'module' } );
+		if ( scriptTag ) {
+			await page.addScriptTag( { content: scriptTag, type: 'module' } );
+		}
 
 		if ( response.status() !== 200 ) {
 			continue;
 		}
 
 		completeRequests++;
 
-		await Promise.all(
-			Object.values( metricsDefinition ).map( async ( value ) => {
-				// Wait until global is populated.
-				await page.waitForFunction(
-					`window.${ value.global } !== undefined`
-				);
+		if ( groupedMetrics.webVitals ) {
+			await Promise.all(
+				Object.values( groupedMetrics.webVitals ).map(
+					async ( value ) => {
+						// Wait until global is populated.
+						await page.waitForFunction(
+							`window.${ value.global } !== undefined`
+						);
+
+						/*
+						 * Do a random click, since only that triggers certain metrics
+						 * like LCP, as only a user interaction stops reporting new LCP
+						 * entries. See https://web.dev/lcp/.
+						 *
+						 * Click off screen to prevent clicking a link by accident and navigating away.
+						 */
+						await page.click( 'body', {
+							offset: { x: -500, y: -500 },
+						} );
+						// Get the metric value from the global.
+						/** @type {number} */
+						const metric = await page.evaluate(
+							( global ) =>
+								/** @type {number} */ window[ global ],
+							value.global
+						);
```

Comment:
This fixes another TypeScript error:

```suggestion
						const metric = /** @type {number} */ await page.evaluate(
							( global ) => window[ global ],
							value.global
						);
```

![image](https://github.com/GoogleChromeLabs/wpp-research/assets/134745/76443de7-d4b2-48b9-920f-06660a98e671)